src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -628,6 +628,73 @@ static void verify(ModuleSymbol m)
         }
     }
 
+    [Fact]
+    public void Operators()
+    {
+        var source = """
+            class C
+            {
+                public static C operator+(ref readonly C x, C y) => x;
+                public static implicit operator C(ref readonly int x) => null;
+                public static explicit operator C(ref readonly int x) => null;
+            }
+            """;
+        CreateCompilation(source).VerifyDiagnostics(
+            // (3,29): error CS0631: ref and out are not valid in this context
+            //     public static C operator+(ref readonly C x, C y) => x;
+            Diagnostic(ErrorCode.ERR_IllegalRefParam, "+").WithLocation(3, 29),
+            // (4,37): error CS0631: ref and out are not valid in this context
+            //     public static implicit operator C(ref readonly int x) => null;
+            Diagnostic(ErrorCode.ERR_IllegalRefParam, "C").WithLocation(4, 37),
+            // (5,37): error CS0631: ref and out are not valid in this context
+            //     public static explicit operator C(ref readonly int x) => null;
+            Diagnostic(ErrorCode.ERR_IllegalRefParam, "C").WithLocation(5, 37),
+            // (5,37): error CS0557: Duplicate user-defined conversion in type 'C'
+            //     public static explicit operator C(ref readonly int x) => null;
+            Diagnostic(ErrorCode.ERR_DuplicateConversionInClass, "C").WithArguments("C").WithLocation(5, 37));
+    }
+
+    [Fact]
+    public void ExpressionTrees()
```

Comment:
>ExpressionTrees [](http://example.com/codeflow?start=16&length=15)

I think we also should test calling methods with `ref readonly` parameters inside an expression tree, With all possible call site modifiers, including none. Also, passing an rValue case.